backend/api/Areas/Tools/Helpers/ImportPropertiesHelper.cs

```diff
@@ -0,0 +1,410 @@
+using Pims.Dal.Services.Admin;
+using Entity = Pims.Dal.Entities;
+
+namespace Pims.Api.Areas.Tools.Helpers
+{
+    /// <summary>
+    /// ImportPropertiesHelper class, provides a way to import properties into the datasource.
+    /// </summary>
+    public class ImportPropertiesHelper
+    {
+        #region Variables
+        private readonly IPimsAdminService _pimsAdminService;
+        private readonly ILogger _logger;
+        private readonly IList<Entity.BuildingConstructionType> _buildingConstructionTypes;
+        private readonly IList<Entity.BuildingPredominateUse> _buildingPredominateUses;
+        private readonly IList<Entity.PropertyClassification> _propertyClassifications;
+        private readonly IList<Entity.PropertyStatus> _propertyStatus;
+        private readonly IList<Entity.City> _cities;
+        private readonly IList<Entity.Agency> _agencies;
+        #endregion
+
+        #region Constructors
+        /// <summary>
+        /// Creates a new instance of a ImportPropertiesHelper class, initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="pimsAdminService"></param>
+        /// <param name="logger"></param>
+        public ImportPropertiesHelper(IPimsAdminService pimsAdminService, ILogger logger)
+        {
+            _pimsAdminService = pimsAdminService;
+            _logger = logger;
+
+            // Preload lookup lists so that they can be references quickly.
+            _buildingConstructionTypes = _pimsAdminService.BuildingConstructionType.GetAll().ToList();
+            _buildingPredominateUses = _pimsAdminService.BuildingPredominateUse.GetAll().ToList();
+            _propertyStatus = _pimsAdminService.PropertyStatus.GetAll().ToList();
+            _propertyClassifications = _pimsAdminService.PropertyClassification.GetAll().ToList();
+            _cities = _pimsAdminService.City.GetAll().ToList();
+            _agencies = _pimsAdminService.Agency.GetAll().ToList();
+        }
+        #endregion
+
+        #region Methods
+        /// <summary>
+        /// Adds or updates the property in the datasource.
+        /// Determines if the property is a parcel or a building.
+        /// Massages some of the data to align with expected values.
+        /// /// </summary>
+        /// <param name="properties"></param>
+        /// <returns></returns>
+        public IEnumerable<Entity.Parcel> AddUpdateProperties(IEnumerable<PropertyModel> properties)
+        {
+            var entities = new List<Entity.Parcel>();
+            foreach (var property in properties)
+            {
+                _logger.LogDebug($"Add/Update property pid:{property.ParcelId}, type:{property.PropertyType}, fiscal:{property.FiscalYear}");
+
+                var valid_pid = int.TryParse(property.ParcelId?.Replace("-", ""), out int pid);
+                if (!valid_pid) continue;
+
+                // Change the agency code.
+                switch (property.Agency)
```

Comment:
At some point we may need to extend this, but for now I think its fine.

Reply:
Hardcode lists are horrible, but for our use-case I wasn't willing to spend additional time.